.gitignore

```diff
@@ -34,3 +34,18 @@ node_modules
 *.iml
 
 .tmp
+
+# npm
+package-lock.json
```

Comment:
I meant to ask about this, I'm guessing this has been excluded due to #42?

Reply:
Yes and no.  On purpose, I exclude in all my projects, the reason being is that NPM did not think it through properly and actually broke optional dependencies, but also, I simply do not see the actual purpose if it.  The package.json file has the ability to "lock" the dependency to a particular version number, which is something is constantly do for production systems so that everyone should always have the same version of the libraries, especially the CI system.  Package.lock is just a way to get around semver, which is a blessing and a curse, but doesn't really add any benefit to a project unless people like to play fast and loose with the versioning.